### DISCUSSION TITLE
Regex: split in multiple lines for documentation

### DISCUSSION THREAD

**Discussion by Finkregh** [renovatebot/renovate#30835] [Suggest an Idea]
### Tell us more.

It would be nice if regexes could be split into multiple lines, that documentation by comments is possible.

Example before:

```json5
    matchStrings: [
        "#\\s*renovate:\\s+datasource=(?<datasource>.*?) depName=(?<depName>.*?) slesVersion=(?<slesVersion>.*?) slesProduct=(?<slesProduct>.*?) slesArchitecture=(?<slesArchitecture>.*?)( versioning=(?<versioning>.*?))?( repoUrl=(?<repoUrl>.*?))?( repoSubdir=(?<repoSubdir>.*?))?\\s\\S+?:(?<currentValue>\\S+)\\s+\\S+:.+",
      ],
```

Example after:

```json5
        matchStringsMultiline: [
            [
                // doc something
                "#\\s*renovate:\\s+datasource=(?<datasource>.*?)",
                // doc
                " depName=(?<depName>.*?)",
                // doc
                " slesVersion=(?<slesVersion>.*?)",
                // doc
                " slesProduct=(?<slesProduct>.*?)",
                // doc
                " slesArchitecture=(?<slesArchitecture>.*?)",
                // doc
                " versioning=(?<versioning>.*?)",
                // doc
                " repoUrl=(?<repoUrl>.*?)",
                // doc
                " repoSubdir=(?<repoSubdir>.*?)",
                // doc
                "\\s\\S+?:(?<currentValue>\\S+)",
                // doc
                "\\s+\\S+:.+",
            ],
		[ ... ]
        ],
```

**Comment by secustor** [comment]
I don't think a separate option is a good idea here. There would be a lot of questions like if there are `matchStrings` and `matchStringsMultine`. Should these be merged or should one take priority. 

The most realistic way I could see this happening is that a nested array is allowed and during config migration it is reduced to a pure string array. In that case if `matchStrings` contains a nested string we can concat it. 

That way we can also still validate the result.

**Reply by secustor** [reply]
For reference that is the current draft PR for the separate config option. https://github.com/renovatebot/renovate/pull/30841

**Reply by Finkregh** [reply]
> In that case if matchStrings contains a nested string we can concat it

I didnt really get that, how would that look in a minimal example config?

**Reply by secustor** [reply]
Custom manager before code migration, so what the content is

```json
{
  "matchStrings": [
     "first string",
     [ 
       "second ",
       "string"
     ]
   ]
}
```

After:

```json
{
  "matchStrings": [
     "first string",
     "second string"
   ]
}
```

**Reply by Finkregh** [reply]
Thanks, dunno when I find time to continue this...
